controllers/common.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dataplanev1 "github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+)
+
+const (
+	// AnsibleSSHPrivateKey ssh private key
+	AnsibleSSHPrivateKey = "ssh-privatekey"
+	// AnsibleSSHPublicKey ssh public key
+	AnsibleSSHPublicKey = "ssh-publickey"
+	// AnsibleSSHAuthorizedKeys authorized keys
+	AnsibleSSHAuthorizedKeys = "authorized_keys"
+)
+
+// ensureSSHSecret - ensures that the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func ensureSSHSecret(
```

Comment:
Isn't this essentially just re-implementing the `lib-common` `VerifySecret` function with some condition handling built in?

Were we not able to set these conditions based on the return value of the lib-common function? Just trying to understand the need to reimplement the logic here.

Reply:
It's little different, other than checking the secret and fields it updates conditions and does not generate hash which we won't use.

Reply:
We can reuse that though. Let me update the PR.

Reply:
Hmm.. I remembered the reason I created a different one, to differentiate between status messages (waiting vs error).